source/storage/apis-1.5.rst

```diff
@@ -338,22 +355,43 @@ collection.
     Posted BSOs whose ids do not appear in either "success" or "failed"
     should be treated as having failed for an unspecified reason.
 
-    Two input formats are available for multiple record POST requests,
-    selected by the *Content-Type* header of the request:
+    To allow upload of large numbers of items while ensuring that other
+    clients do not sync down inconsistent data, this API can combine several
+    POST requests into a single "batch" so that all modified BSOs appear to
+    have been submitted at the same time.  Batching behaviour is controlled
+    by the following query parameters:
 
-    - **application/json**: the input is a JSON list of objects, one for
-      for each BSO in the request.
+    - **batch**: indicates that uploads should be batched together into a
+      single conceptual update.  To begin a new batch pass the string 'true'.
+      To add more items to an existing batch pass a previously-obtained batch
+      identifier.
 
-    - **application/newlines**: each BSO is sent as a separate JSON object
-      followed by a newline.
+    - **commit**: indicates that the batch should be commited, and all items
+      uploaded to that batch made visible to other clients.  If present, it
+      must be the string 'true' and be combined with a batch identifier.
 
-    For backwards-compatibility with existing clients, the server will also
-    treat **text/plain** input as JSON.
+    When submitting items for inclusion in a multi-request batch upload,
+    successful responses will have a "202 Accepted" status code, and will
+    contain a JSON object giving the batch identifier rather than modification
+    time.
+
+    For example::
+
+        {
+         "batch": "OPAQUEBATCHID",
+         "success": ["GXS58IDC_12", "GXS58IDC_13", "GXS58IDC_15",
+                     "GXS58IDC_16", "GXS58IDC_18", "GXS58IDC_19"],
+         "failed": {"GXS58IDC_11": "invalid ttl"],
+                    "GXS58IDC_14": "invalid sortindex"}
```

Comment:
> Presumably failures here will only be in record parsing, not in storage, right?

Right.  The idea would be that if we've reported success at this stage, it'll only fail if the entire batch fails.

> do success and failure lists appear in the intermediate batch responses

I think they should, so that we can report individual item failures to parse etc.  What I *don't* think we should do is the current behavior were we tell you that the first 100 items succeeded, but all the others were rejected because you sent too many.  That should just fail the entire request when in batch mode.  (It *should* fail the entire request in non-batch mode to, but that would risk preventing old clients who send such items from making progress)